packages/sanity/src/core/form/useDocumentForm.ts

```diff
@@ -259,12 +267,21 @@ export function useDocumentForm(options: DocumentFormOptions): DocumentFormValue
   )
 
   const requiredPermission = value._createdAt ? 'update' : 'create'
+  const getDocumentId = useCallback(
+    (docId: string) => {
+      if (liveEdit || isPublishedId(docId)) {
+        return getPublishedId(docId)
+      }
+      return isDraftId(docId) ? getDraftId(docId) : docId
```

Comment:
this can be simplified to
```suggestion
      return docId
```